avro-migration-helper/build.gradle

```diff
@@ -15,22 +31,77 @@ dependencies {
     exclude group: "org.jboss.netty"
     exclude group: "com.thoughtworks.paranamer", module: "paranamer-ant"
   }
-  //when run with #avro17
-  //compileOnly 'org.apache.avro:avro-compiler:1.7.7'
 
-  compile 'org.json:json:20170516'
+  //comment above and uncomment these when trying to
+  //work on a particular version of avro in an IDE
+
+  //compile 'org.apache.avro:avro:1.8.2'
+  //compile 'org.apache.avro:avro-compiler:1.8.2'
+
+  compile 'org.json:json:20180813'
 
-  testCompile ('org.apache.avro:avro:1.4.1') {
+  testCompileOnly ('org.apache.avro:avro:1.4.1') {
     exclude group: "org.mortbay.jetty"
     exclude group: "org.apache.velocity"
     exclude group: "commons-lang"
     exclude group: "org.jboss.netty"
     exclude group: "com.thoughtworks.paranamer", module: "paranamer-ant"
   }
-  //when run with #avro17
-  //compileOnly 'org.apache.avro:avro-compiler:1.7.7'
+
   testCompile 'org.testng:testng:6.14.3'
   testCompile 'commons-io:commons-io:2.6'
   testCompile 'org.skyscreamer:jsonassert:1.5.0'
   testCompile 'net.openhft:compiler:2.3.1'
-}
\ No newline at end of file
+
+  testRuntimeOnly 'org.slf4j:slf4j-simple:1.7.26'
+
+  avro14 ("org.apache.avro:avro:1.4.1") {
+    exclude group: "org.mortbay.jetty"
+    exclude group: "org.apache.velocity"
+    exclude group: "commons-lang"
+    exclude group: "org.jboss.netty"
+    exclude group: "com.thoughtworks.paranamer", module: "paranamer-ant"
+    exclude group: "org.slf4j"
+  }
+  avro15 "org.apache.avro:avro:1.5.4"
+  avro15 "org.apache.avro:avro-compiler:1.5.4"
+  avro16 "org.apache.avro:avro:1.6.3"
+  avro16 "org.apache.avro:avro-compiler:1.6.3"
+  avro17 ("org.apache.avro:avro:1.7.7") {
+    exclude group: "org.slf4j"
+  }
+  avro17 ("org.apache.avro:avro-compiler:1.7.7") {
+    exclude group: "org.slf4j"
+  }
+  avro18 ("org.apache.avro:avro:1.8.2") {
+    exclude group: "org.slf4j"
+  }
+  avro18 ("org.apache.avro:avro-compiler:1.8.2") {
+    exclude group: "org.slf4j"
+  }
+}
+
+def avroVersions = ["14", "15", "16", "17", "18"]
+
+for (String avroVersion : avroVersions) {
+
+  task "testAvro${avroVersion}" (type: Test, group: "Verification", description: "runs unit tests with avro ${avroVersion}") {
+    useTestNG()
+
+    testLogging {
+      events "passed", "skipped", "failed"
+    }
+
+    classpath = project.sourceSets.test.runtimeClasspath.filter {
+      File file -> !file.name.contains("avro")
+    }.plus (configurations."avro${avroVersion}")
+
+    doFirst {
+      System.out.println("classpath for avro ${avroVersion} is:")
```

Comment:
_something... tells me this doesn't work right_
_something_

Reply:
will fix :+1:  :-)